src/Microsoft.DotNet.ProjectJsonMigration/ProjectMigrator.cs

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.Build.Evaluation;
+using Microsoft.Build.Construction;
+using Microsoft.DotNet.ProjectModel;
+using Microsoft.DotNet.Cli;
+using System.Linq;
+using System.IO;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.DotNet.ProjectJsonMigration
+{
+    public class ProjectMigrator
+    {
+        // TODO: Migrate PackOptions
+        // TODO: Support Mappings in IncludeContext Transformations
+        // TODO: Migrate Multi-TFM projects
+        // TODO: Tests
+        // TODO: Out of Scope
+        //     - Globs that resolve to directories: /some/path/**/somedir
+        //     - Migrating Deprecated project.jsons
+        //     - Configuration dependent source exclusion
+
+        public void Migrate(MigrationSettings migrationSettings)
+        {
+            var projectDirectory = migrationSettings.ProjectDirectory;
+            EnsureDirectoryExists(migrationSettings.OutputDirectory);
+
+            var migrationRuleInputs = ComputeMigrationRuleInputs(migrationSettings);
+            VerifyInputs(migrationRuleInputs);
```

Comment:
Should "VerifyInputs" go first?  Does it make sense to do the other work if the inputs are invalid?